src/containers/PaymentInfo.tsx

```diff
@@ -0,0 +1,178 @@
+import React, { useState } from 'react';
+import {
+	View,
+	Text,
+	StyleSheet,
+	TextInput,
+	Dimensions,
+	TouchableOpacity,
+	TouchableWithoutFeedback,
+	Keyboard,
+	KeyboardAvoidingView,
+	Alert,
+} from 'react-native';
+import { StackNavigationProp } from '@react-navigation/stack';
+import { RootStackParamList } from '../../App';
+
+import Loading from '../components/Loading';
+
+interface PaymentInfoProps {
+	navigation: StackNavigationProp<RootStackParamList, 'PaymentInfo'>;
+}
+
+const screenWidth = Dimensions.get('window').width;
+const screenHeight = Dimensions.get('window').height;
+
+const PaymentInfo = ({ navigation }: PaymentInfoProps) => {
+	const [cardNumber, setCardNumber] = useState('');
+	const [expiryDate, setExpiryDate] = useState('');
+	const [cvv, setCVV] = useState('');
+	const [loading, setLoading] = useState(false);
+
+	const cleanupCardNumber = (number: string) => {
+		let cleanup = number
+			.replace(/\s?/g, '')
+			.replace(/(\d{4})/g, '$1 ')
+			.trim();
+		if (cleanup.slice(-1) === '.')
+			cleanup = cleanup.substring(0, cleanup.length - 1);
+		setCardNumber(cleanup);
+	};
+	const cleanupExpiryDate = (number: string) => {
+		if (number.slice(-1) === '.')
+			number = number.substring(0, number.length - 1);
+		else if (number.length === 2 && expiryDate.length == 1) number += '/';
+		else if (number.length === 2 && expiryDate.length == 3)
+			number = number.substring(0, number.length - 1);
+		setExpiryDate(number);
+	};
+	const processAndSubmitPayment = () => {
+		if (cardNumber.length !== 19) {
+			setLoading(false);
+			Alert.alert('Invalid Card Number');
+			return;
+		}
+		if (
+			expiryDate.length !== 5 ||
+			parseInt(expiryDate.substring(0, 2)) <= 0 ||
+			parseInt(expiryDate.substring(0, 2)) > 12
+		) {
+			setLoading(false);
+			Alert.alert('Invalid Expiry Date');
+			return;
+		}
+		if (cvv.length !== 3) {
+			setLoading(false);
+			Alert.alert('Invalid CVV');
+			return;
+		}
+
+		// TODO: Stripe API Payment Processing
+		setTimeout(() => {
+			setLoading(false);
+			console.log('Success');
+			navigation.navigate('PaymentSuccess', {
+				success: true,
+			});
+		}, 5000);
+	};
+
+	return (
+		<TouchableWithoutFeedback onPress={() => Keyboard.dismiss()}>
+			<KeyboardAvoidingView behavior="position" style={styles.container}>
```

Comment:
I just realized I wrote a `Form` component for this exact purpose of doing the keyboard avoiding and all that jazz. Maybe we would be able to use it here? But let's just add a todo for now

Reply:
I don't see the Form component, so for now I've left a TODO to change KeyboardAvoidingView to use your form component